whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageDequeue.java

```diff
@@ -195,17 +195,21 @@ private void handleMessage(final String messageId) {
             final EmailMessageInfo bouncedMessage = messageService.getBouncedMessageInfo(message);
             if (bouncedMessage != null) {
                 messageService.verifyAndSetAsUndeliverable(bouncedMessage);
-                mailMessageDao.deleteMessage(messageId);
                 return;
             }
 
             final EmailMessageInfo unsubscribeMessage = messageService.getUnsubscribedMessageInfo(message);
             if (unsubscribeMessage != null) {
                 messageService.verifyAndSetAsUnsubscribed(unsubscribeMessage);
-                mailMessageDao.deleteMessage(messageId);
                 return;
             }
+        } catch (Exception e){
+            LOGGER.error("Error detecting bounce detection or unsubscribing", e);
```

Comment:
I am not so sure if you want  to delete message for every type of exception

Reply:
yes, lets wait for Ed. Maybe just for parsing issues but I am not sure. I will put this as a draft until we know when we need to remove the mail

Reply:
Yes always clean up (delete) the message, as we don't want these DSN or unsubscribe responses to get stuck. Still log an ERROR because it's unexpected.

Reply:
But don't always delete every kind of message! Only bounce and unsubscribe messages

Reply:
you can have an error parsing unsubscribe/bounce messages and don't know if the message is unsubscribed or bounced